sds/src/scanner/error.rs

```diff
@@ -45,6 +49,55 @@ pub enum MatchValidationError {
     NoMatchValidationType,
 }
 
+#[derive(Debug, PartialEq, Eq, Error)]
+pub enum ScannerMetadataError {
```

Comment:
As these errors are not used in the dd-sensitive-data-scanner code. I wonder if it would be possible to move it to internal library with a different error type. It would help to split the ownership between internal library and public engine.